dcc-portal-ui/app/scripts/ui/js/table.js

```diff
@@ -302,7 +326,49 @@ angular.module('icgc.ui.table.sortable', []).directive('sortable', function ($lo
   };
 });
 
+angular.module('icgc.ui.table.filter', [])
+  .directive('tableFilter', function(gettextCatalog){
+    return {
+      restrict: 'E',
+      scope: {
+        filterModel: '=',
+        class: '@'
+      },
+      template: '<span class="t_suggest t_suggest__header table-filter {{class}}">' +
+        '<input type="text" class="t_suggest__input form-control" placeholder="' + gettextCatalog.getString('Table filter') + 
+        '" data-ng-model="filterModel" />' + 
+        '<i class="t_suggest__embedded t_suggest__embedded__left t_suggest__embedded__search icon-search">' +
+        '</i>'+
+        '<i class="t_suggest__embedded t_suggest__embedded__right t_suggest__embedded__clear icon-cancel ng-hide"' + 
+        ' data-ng-click="filterModel = \'\'" data-ng-show="filterModel"></i>' +
+        '</span>',
+      replace: true
```

Comment:
Not directed at you Sid, but we really need to get es6 template strings in here
This multiline template is a bit of a headache to look at but we don't really have options with current tooling